pyiceberg/view/metadata.py

```diff
@@ -0,0 +1,89 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+from __future__ import annotations
+
+from typing import Dict, List, Literal, Optional
+
+from pydantic import Field, field_validator
+
+from pyiceberg.schema import Schema
+from pyiceberg.typedef import IcebergBaseModel, Properties
+from pyiceberg.types import transform_dict_value_to_str
+
+
+class SQLViewRepresentation(IcebergBaseModel):
+    """Represents the SQL query that defines the view."""
+
+    type: Literal["sql"] = Field()
+    """A string that indicates the type of representation. Must be `sql`"""
+    sql: str = Field()
+    """A string that contains the SQL text of the view definition."""
+    dialect: str = Field()
+    """The dialect of the SQL, e.g. `spark`, `trino`, `presto`."""
+
+
+class ViewVersion(IcebergBaseModel):
+    """A version of the view definition."""
+
+    version_id: int = Field(alias="version-id")
+    """ID for the version"""
+    schema_id: int = Field(alias="schema-id")
+    """ID of the schema for the view version"""
+    timestamp_ms: int = Field(alias="timestamp-ms")
+    """Timestamp when the version was created (ms from epoch)"""
+    summary: Dict[str, str] = Field()
+    """A string to string map of summary metadata about the version"""
+    representations: List[SQLViewRepresentation] = Field()
+    """A list of representations for the view definition"""
+    default_catalog: Optional[str] = Field(alias="default-catalog", default=None)
+    """Catalog name to use when a reference in the SELECT does not contain a catalog"""
+    default_namespace: List[str] = Field(alias="default-namespace")
```

Comment:
We should use the existing Namespace class here instead

```suggestion
    default_namespace: Namespace = Field(alias="default-namespace")
```

Reply:
Should this be Identifier instead? Namespace doesn't appear to be a valid class.